plugins/kubernetes/app/models/watchers/cluster_pod_error_watcher.rb

```diff
@@ -2,34 +2,38 @@ module Watchers
   class ClusterPodErrorWatcher < BaseClusterWatcher
     include Celluloid::Notifications
 
-    def initialize(client)
-      @client = client
-      super(@client.watch_events(field_selector: 'involvedObject.kind=Pod'))
+    attr_reader :cluster
+
+    def initialize(cluster)
+      super(cluster.client.watch_events(field_selector: 'involvedObject.kind=Pod'))
+      @cluster = cluster
     end
 
     private
 
     def handle_notice(notice)
-      if error_notice?(notice.object.reason)
-        project_name = project_name(notice.object.involvedObject)
-        publish(project_name, notice) if project_name
+      event = Events::ClusterEvent.new(notice)
+
+      if pod_error?(event)
+        pod = get_pod(event.involved_object)
+        publish("pod-events-project-#{pod.project_id}", message(event, pod: pod)) unless pod.nil?
       end
     end
 
-    def project_name(involved_object)
-      pod = @client.get_pod(involved_object.name, involved_object.namespace)
-      pod.metadata.labels ? pod.metadata.labels.project : nil
+    def get_pod(involved_object)
+      api_pod = cluster.client.get_pod(involved_object.name, involved_object.namespace)
+      Kubernetes::Api::Pod.new(api_pod)
     rescue KubeException => e
-      if e.error_code == 404
```

Comment:
Checking for the 404 was done to handle events for deleted pods where we could not get the pod through the `get_pod` call anymore. Now it seems we ignore all errors coming from the API.